be/src/vec/exprs/lambda_function/varray_match_function.cpp

```diff
@@ -0,0 +1,188 @@
+                    assert_cast<const ColumnNullable*>(first_column.get())->get_nested_column_ptr();
+            const auto& column_array_nullmap =
+                    assert_cast<const ColumnNullable*>(first_column.get())->get_null_map_column();
+            VectorizedUtils::update_null_map(first_outside_null_map->get_data(),
+                                             column_array_nullmap.get_data());
+        }
+        const ColumnArray& first_col_array = assert_cast<const ColumnArray&>(*first_arg_column);
+        const auto& first_off_data =
+                assert_cast<const ColumnArray::ColumnOffsets&>(first_col_array.get_offsets_column())
+                        .get_data();
+        const auto& first_nested_nullable_column =
+                assert_cast<const ColumnNullable&>(*first_col_array.get_data_ptr());
+
+        // result is nullable bool column for every array column
+        auto result_data_column = ColumnUInt8::create(input_rows, 1);
+        auto result_null_column = ColumnUInt8::create(input_rows, 0);
+
+        auto second_arg_column = second_column;
+        auto second_outside_null_map = ColumnUInt8::create(input_rows, 0);
+        if (second_arg_column->is_nullable()) {
+            second_arg_column = assert_cast<const ColumnNullable*>(second_column.get())
+                                        ->get_nested_column_ptr();
+            const auto& column_array_nullmap =
+                    assert_cast<const ColumnNullable*>(second_column.get())->get_null_map_column();
+            VectorizedUtils::update_null_map(second_outside_null_map->get_data(),
+                                             column_array_nullmap.get_data());
+        }
+        const ColumnArray& second_col_array = assert_cast<const ColumnArray&>(*second_arg_column);
+        const auto& second_off_data = assert_cast<const ColumnArray::ColumnOffsets&>(
+                                              second_col_array.get_offsets_column())
+                                              .get_data();
+        const auto& second_nested_null_map_data =
+                assert_cast<const ColumnNullable&>(*second_col_array.get_data_ptr())
+                        .get_null_map_column()
+                        .get_data();
+        const auto& second_nested_column =
+                assert_cast<const ColumnNullable&>(*second_col_array.get_data_ptr())
+                        .get_nested_column();
+        const auto& second_nested_data =
+                assert_cast<const ColumnUInt8&>(second_nested_column).get_data();
+
+        //3. calculate result column with second bool column
+        for (int row = 0; row < input_rows; ++row) {
+            //first or second column is null, so current row is invalid data
+            if (first_outside_null_map->get_data()[row] ||
+                second_outside_null_map->get_data()[row]) {
+                result_null_column->get_data()[row] = 1;
+                result_data_column->get_data()[row] = 0;
```

Comment:
better add comment to explain why not return null instead of false, and need to describe the behavior in document